docs/source/app_developers_guide/docker.rst

```diff
@@ -0,0 +1,416 @@
+  sawtooth-validator-default         | [00:27:56.762 INFO     publisher] Now building on top of block: 44ccc3e6(1, S:910b9c23, P:05b2a651)
+  sawtooth-validator-default         | [00:27:56.763 INFO     chain] Finished block validation of: 44ccc3e6(1, S:910b9c23, P:05b2a651)
+  Gracefully stopping... (press Ctrl+C again to force)
+  Stopping sawtooth-tp_xo_python-default ... done
+  Stopping sawtooth-tp_config-default ... done
+  Stopping sawtooth-client-default... done
+  Stopping sawtooth-rest_api-default ... done
+  Stopping sawtooth-tp_intkey_python-default ... done
+  Stopping sawtooth-validator-default ... done
+
+  % docker-compose -f sawtooth-default.yaml down
+
+
+
+.. note::
+
+  Genesis block, validator and transaction processor startup are handled for you
+  if you're using the Docker workflow. If you want a better understanding of
+  this process, see :doc:`ubuntu`.
+
+
+Confirming Connectivity 
+=======================
+
+To confirm that a validator is up and running, and reachable from the client
+container, you can use the curl command:
+
+.. code-block:: console
+
+	root@75b380886502:/# curl http://rest_api:8080/blocks
+
+If the validator is running and reachable, the output should be similar to:
+
+.. code-block:: console
+
+	{
+	  "data": [
+	    {
+	      "batches": [],
+	      "header": {
+	        "batch_ids": [],
+	        "block_num": 0,
+	        "consensus": "R2VuZXNpcw==",
+	        "previous_block_id": "0000000000000000",
+	        "signer_pubkey": "03061436bef428626d11c17782f9e9bd8bea55ce767eb7349f633d4bfea4dd4ae9",
+	        "state_root_hash": "708ca7fbb701799bb387f2e50deaca402e8502abe229f705693d2d4f350e1ad6"
+	      },
+	      "header_signature": "119f076815af8b2c024b59998e2fab29b6ae6edf3e28b19de91302bd13662e6e43784263626b72b1c1ac120a491142ca25393d55ac7b9f3c3bf15d1fdeefeb3b"
+	    }
+	  ],
+	  "head": "119f076815af8b2c024b59998e2fab29b6ae6edf3e28b19de91302bd13662e6e43784263626b72b1c1ac120a491142ca25393d55ac7b9f3c3bf15d1fdeefeb3b",
+	  "link": "http://rest_api:8080/blocks?head=119f076815af8b2c024b59998e2fab29b6ae6edf3e28b19de91302bd13662e6e43784263626b72b1c1ac120a491142ca25393d55ac7b9f3c3bf15d1fdeefeb3b",
+	  "paging": {
+	    "start_index": 0,
+	    "total_count": 1
+	  }
+	}root@75b380886502:/#
+
+If the validator process, or the validator container, is not running, the curl
+command will return nothing, or time out.
+
+
+Transaction Processors
+======================
+
+The Docker Compose file (*sawtooth-default.yaml*) starts the following
+transaction processors automatically:
+
+* tp_config
+* tp_intkey_python
+* tp_xo_python
+
+These transaction processors handle transactions for the config, intkey, and XO transaction families, and are written in Python.
+
+
+Creating And Submitting Transactions
+====================================
+
+The **intkey** command is provided to create sample transactions of the intkey
+transaction type for testing purposes.
```

Comment:
The **intkey** command can be used to create and submit IntegerKey transactions. The transactions created conform to the :doc:`/transaction_family_specifications/integerkey_transaction_family`